st2actions/st2actions/runners/httprunner.py

```diff
@@ -172,37 +173,35 @@ def __init__(self, url=None, method=None, body='', params=None, headers=None, co
     def run(self):
         results = {}
         resp = None
+        json_content = self._is_json_content()
+
         try:
-            if self._is_json_content():
-                # use the json property instead of data in case of json content
-                # also cast property to a dict.
-                resp = requests.request(
-                    self.method,
-                    self.url,
-                    params=self.params,
-                    json=self._cast_object(self.body),
-                    headers=self.headers,
-                    cookies=self.cookies,
-                    auth=self.auth,
-                    timeout=self.timeout,
-                    allow_redirects=self.allow_redirects,
-                    proxies=self.proxies,
-                    files=self.files
-                )
+            if json_content:
+                # cast params (body) to dict
+                data = self._cast_object(self.body)
+
+                try:
+                    data = json.dumps(data)
+                except ValueError:
+                    msg = 'Request body (%s) can\'t be parsed as JSON' % (data)
+                    raise ValueError(msg)
             else:
-                resp = requests.request(
-                    self.method,
-                    self.url,
-                    params=self.params,
-                    data=self.body,
-                    headers=self.headers,
-                    cookies=self.cookies,
-                    auth=self.auth,
-                    timeout=self.timeout,
-                    allow_redirects=self.allow_redirects,
-                    proxies=self.proxies,
-                    files=self.files
-                )
+                data = self.body
+
+            resp = requests.request(
+                self.method,
+                self.url,
+                params=self.params,
+                data=data,
```

Comment:
well I was also using the json property. Are you sure that does not matter much? Maybe it is only to do with setting the right headers?